lib/dns/server.js

```diff
@@ -395,6 +398,27 @@ class RootServer extends DNSServer {
     const {name, type} = qs;
     const tld = util.from(name, -1);
 
+    // Plugins can insert middleware here and hijack the
+    // lookup for special TLDs before checking Urkel tree.
+    // We also pass the entire question in case a plugin
+    // is able to return an authoritative (non-referral) answer.
+    if (typeof this.middle === 'function') {
```

Comment:
Maybe hook or resolveHook would be more intuitive name?

Reply:
Hmmm, appreciate the feedback. I've already written two plugins using this and documented the `middle()` property here: https://hsd-dev.org/guides/plugins.html 🤷‍♂️